src/osd/modules/sound/sdl_sound.cpp

```diff
@@ -97,13 +97,18 @@ int sound_sdl::init(osd_interface &osd, const osd_options &options)
 	// Capture is not implemented in SDL2, and the enumeration
 	// interface is different in SDL3
 	int dev_count = SDL_GetNumAudioDevices(0);
+#if 0
 	for(int i=0; i != dev_count; i++) {
+#else
+	for(int i=dev_count-1; i != -1; i--) {
+#endif
 		SDL_AudioSpec spec;
 		const char *name = SDL_GetAudioDeviceName(i, 0);
 		int err = SDL_GetAudioDeviceSpec(i, 0, &spec);
 		if(!err)
 			m_devices.emplace_back(name, spec.freq, spec.channels);
 	}
+#if SDL_MAJOR_VERSION > 2 || SDL_MINOR_VERSION >= 24
```

Comment:
You’re supposed to use `#if SDL_VERSION_ATLEAST(x, y, z)` for these kinds of things.